client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -92,34 +92,67 @@ class StatsPostDetail extends Component {
 			return translate( 'Home page / Archives' );
 		}
 
-		if ( typeof post?.title === 'string' && post.title.length ) {
+		if ( typeof post?.title === 'string' && post?.title.length ) {
 			return decodeEntities( stripHTML( post.title ) );
 		}
 
-		if ( typeof postFallback?.post_title === 'string' && postFallback.post_title.length ) {
+		if ( typeof postFallback?.post_title === 'string' && postFallback?.post_title.length ) {
 			return decodeEntities( stripHTML( postFallback.post_title ) );
 		}
 
 		return null;
 	}
 
+	getPost() {
+		const { isPostHomepage, post, postFallback } = this.props;
+
+		if ( typeof post === 'object' && post?.title.length ) {
+			return {
+				title: this.getTitle(),
```

Comment:
I'm wondering as we check for `title` here too, so is it still necessary for the function `this.getTitle()` ? Another thing I noticed is that `'/sites/%d/posts/(?P<resource_id>[\d]+)'` is available on Odyssey, and I can see `QueryPostStats`, seems something fishy happens in the reducer?
https://github.com/Automattic/wp-calypso/blob/c50b21a4a0716cfc2ebb4d424981b005da1e526a/client/state/posts/reducer.js#L116

Reply:
I noticed the response from endpoint `/sites/${site-id}/posts/${post-id}` on Odyssey Stats missed most details compared to Calypso Stats. This change will resolve the missing title shallowly, and we can patch missed data for the [expected post](https://github.com/Automattic/wp-calypso/blob/9eaa3ca071f24b2fd0486900c206513b7b00ba77/client/my-sites/stats/stats-post-detail/index.jsx#L238).